### PR TITLE
ConfigManger and logger classes 

### LINKED ISSUE #8: Install or create a logger in order to properly display information in the console.
A logger makes it possible to display the time, the type of message, to color errors, and even to record errors in a file. 

We need to set one up.
**There are several, but I've listed four of them :** 
- ``Node-Loggly``
- ``Bunyan``
- ``Winston``
- ``Morgan``

You can find more information on how they work and what they do here: [https://www.loggly.com/blog/node-js-libraries-make-sophisticated-logging-simpler/](https://www.loggly.com/blog/node-js-libraries-make-sophisticated-logging-simpler/)

**I propose to use Winston.**
**Winston includes the following features:** 

- **Streams:** Winston uses Node.js streams, which is a performant way of chunking up data processing—for instance, data manipulation on large amounts of data. Streams break tasks down into small, manageable chunks so it’s easier for your Node.js system to handle. Curious why? Read up on streams and the event loop.

- **Transports:** Logs need to go somewhere. By default, console.log goes to your standard output file. However, you may also want to stream your logs to a central location, or store them in a database for querying later. Winston allows you to do that easily with a concept called transports. What’s also nice about using Winston is that you can have as many transports as you like.

- **Formatting:** Your regular console.log, by default, only deals in strings. We’ll come back to why this is problematic later. But Winston has some advanced features for formatting log code before stashing it away. For example, Winston offers JSON formatting, coloring log output, and the ability to fiddle with formats before they’re posted off to your end log locations.

- **Levels:** Winston adds another interesting layer, which is log levels. Log levels are metadata for logs. In plain English, this translates to data about data. Log levels tell you the severity, ranging from as severe as a system outage to as small as a deprecated function. The reason for having log levels in the first place is so you can quickly see which logs need attention. Log levels also allow you to configure environments to log the correct amount of data, with amounts of detail.

### PR DESCRIPTION
**Pull request by SirMishaa**
**ConfigManager** is a class that allows you to read configuration files. For now, it reads the file ``src/config/botconfig.json``. 

**This file contains the configuration for the moderator and administrator roles.**

In adequacy with issue #8 , this pull request will also contain **the log system (Winston)** in order to display the information correctly in the console.

### CODE REVIEW

File: src/Core/ConfigManager.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    }
+
+    /**
+     * This function will load config from Botconfig.json
+     */
+    public loadConfig(): void {
+        this.moderatorRole = configData.roles.moderator;
+        this.administratorRole = configData.roles.administrator;
+    }
+
```
Comment (qcarpentier) [inline src/Core/ConfigManager.ts:35]: What could be the utility of `saveConfig`? I don't completely get it. @ASmallCookie

Comment (SirMishaa) [reply src/Core/ConfigManager.ts:35]: ``saveConfig`` is a function which save changes made, => in the **botconfig.json** file.

I plan to use it that way : 
```ts
let configManager = ConfigManager.getInstance();
let config = configManager.getConfig();
configManager.saveConfig(...config, {logger.saveInFile = true, logger.saveFilename = "hello.log"});
````
It's just an example
@qcarpentier

**Comment by SirMishaa** [comment]
Take a look at the last commit, and tell me what you think about it.
a522d59

**Comment by qcarpentier** [comment]
@ASmallCookie 
Is the pull request ready for review? Because the _WIP label_ is still there, keep me updated 😜

**Comment by SirMishaa** [comment]
> 
> 
> @ASmallCookie
> Is the pull request ready for review? Because the _WIP label_ is still there, keep me updated 😜

Not yet, I'll remove the WIP label when it will be.
They stay just few things to do.

**Review by qcarpentier — CHANGES_REQUESTED** [review]
As you wish

**Review by SirMishaa — COMMENTED** [review]
Fixed
